wallet/src/main/scala/org/bitcoins/wallet/LockedWallet.scala

```diff
@@ -161,10 +182,186 @@ abstract class LockedWallet extends LockedWalletApi with BitcoinSLogger {
+          }
+
+          def processUtxo(index: Int): Future[SpendingInfoDb] =
+            addUtxo(transaction, UInt32(index))
+              .flatMap {
+                case AddUtxoSuccess(utxo) => Future.successful(utxo)
+                case err: AddUtxoError =>
+                  logger.error(s"Could not add UTXO", err)
+                  Future.failed(err)
+              }
+
+          val addUTXOsFut: Future[Seq[(SpendingInfoDb, OutputWithIndex)]] =
+            Future
+              .sequence {
+                xs.map(out => processUtxo(out.index).map(_ -> out))
+              }
+
+          val createIncomingTxo: Seq[(SpendingInfoDb, OutputWithIndex)] => Vector[
+            IncomingWalletTXO] =
+            _.map {
+              case (utxo, OutputWithIndex(out, index)) =>
+                IncomingWalletTXO(
+                  confirmations = confirmations,
+                  // is this always the case?
+                  spent = false,
+                  scriptPubKey = out.scriptPubKey,
+                  txid = transaction.txIdBE,
+                  // always defined, as its freshly
+                  // written to the DB
+                  spendingInfoID = utxo.id.get
+                )
+            }.toVector
+
+          val incomingTXOsFut = addUTXOsFut.map(createIncomingTxo)
+          val writeIncomingTXOsFut =
+            incomingTXOsFut.flatMap(incomingTxoDAO.createAll)
+
+          writeIncomingTXOsFut
+
+      }
+    }
+  }
+
+  /**
+    * Processes an incoming transaction that already exists in our wallet.
+    * If the incoming transaction has more confirmations than what we
+    * have in the DB, we update the TX
+    */
+  private def processExistingIncomingTxo(
+      transaction: Transaction,
+      confirmations: Int,
+      foundTxo: IncomingWalletTXO): Future[Option[IncomingWalletTXO]] = {
+    if (foundTxo.confirmations < confirmations) {
+      logger.debug(
+        s"Increasing confirmation count of txo=${transaction.txIdBE}, old=${foundTxo.confirmations} new=${confirmations}")
+      val updateF =
+        incomingTxoDAO.update(foundTxo.copy(confirmations = confirmations))
+
+      updateF.foreach(tx =>
+        logger.debug(
+          s"Updated confirmation count=${tx.confirmations} of output=${foundTxo}"))
+      updateF.failed.foreach(err =>
+        logger.error(
+          s"Failed to update confirmation count of transaction=${transaction.txIdBE}",
+          err))
+
+      updateF.map(Some(_))
+    } else if (foundTxo.confirmations > confirmations) {
+      val msg =
+        List(
+          s"Incoming transaction=${transaction.txIdBE} has fewer confirmations=$confirmations",
+          s"than what we already have registered=${foundTxo.confirmations}! I don't know how",
+          s"to handle this."
+        ).mkString(" ")
+      logger.warn(msg)
+      Future.failed(new RuntimeException(msg))
+    } else {
+      logger.debug(
```

Comment:
This could be bad in the case of competing re-orgs. Do we really want this to be our behavior?

Reply:
Reorg isn't handled any where in the wallet currently. I can add a comment here, saying that this will need to change once reorg handling is implemented.